### DISCUSSION TITLE
PdfOptions configuration parameters do not take effect

### DISCUSSION THREAD

**Discussion by xiazhao001** [hardkoded/puppeteer-sharp#2843] [Q&A]
PuppeteerSharp:17.0.0 

Print configuration parameters:

  PdfOptions pdfOptions = new PdfOptions();
  pdfOptions.PrintBackground = true;
  pdfOptions.Format = PaperFormat.A4;
  pdfOptions.PreferCSSPageSize = true;
  pdfOptions.DisplayHeaderFooter = true; 
  pdfOptions.FooterTemplate = "Footer Text";   
  pdfOptions.Format = new PuppeteerSharp.Media.PaperFormat(8.27m, 11.69m);  
  pdfOptions.HeaderTemplate = "";   
  pdfOptions.Landscape = false;     
  pdfOptions.MarginOptions = new PuppeteerSharp.Media.MarginOptions() { Bottom = "0px", Left = "0px", Right = "0px", Top = "0px" }; 
  pdfOptions.Scale = 1m;

**Comment by kblok** [comment]
What's not taking effect?

**Comment by xiazhao001** [comment]
@kblok Setting margins in PDF printing properties is not effective

**Comment by Harshil1088** [comment]
For me also setting margin not working. In next pdf page, table content starts from top only and not setting any margin. Is any one have fix for it? @kblok

**Reply by kblok** [reply]
Do you have some code I can use to test?

**Reply by xiazhao001** [reply]
I upgraded to 18.1.0, which solved my problem @Harshil1088
